### PR TITLE
project-19-shaun-newmark

### PR DESCRIPTION
**Pull request by newmarkdev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 ## The problem
 
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+Users are able to post a message and like old messages with a like button. The API is linked to MongoDB. 
 
 ## View it live
 
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+API: https://happy-thoughts-api-sn.herokuapp.com/
+
```
Comment (AsaHildebrand) [inline README.md:12]: Maybe could add a short documentation of the endpoints 😃

File: server.js [hunk 2]
```diff
@@ -1,29 +1,70 @@
...
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+app.get('/thoughts', async (req, res) => {  
+  const allThoughts = await Thought.find().sort({ CreatedAt: -1 }).limit(20)
+  res.json(allThoughts)
+})
+
+app.post('/thoughts', async (req, res) => {
```
Comment (AsaHildebrand) [inline server.js:45]: Posting thoughts works fine :)

File: server.js [hunk 3]
```diff
@@ -1,29 +1,70 @@
...
+    res.json(newThought)
+  } catch (error) {
+    if (error.code === 11000) {
+      res.status(400).json({ error: 'Duplicated value', fields: error.keyValue })
+    }
+    res.status(400).json(error)
+  }
+})
+
+app.post('/thoughts/:id/likes', async (req, res) => {
```
Comment (AsaHildebrand) [inline server.js:57]: To like thoughts also works fine :)

**Review by AsaHildebrand — APPROVED** [review]
Nice API, everything works fine and the most recent 20 thoughts is shown 🥇 Don't have anything to add or change. Nice job! 😃

File: server.js [hunk 4]
```diff
@@ -1,29 +1,70 @@
...
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String, 
+    required: true,    
+    minlength: 5,
+    maxlength: 140
```
Comment (KarolCislowski) [inline server.js:19]: nicely configured field!

File: server.js [hunk 5]
```diff
@@ -1,29 +1,70 @@
...
+    }
+    res.status(400).json(error)
+  }
+})
+
+app.post('/thoughts/:id/likes', async (req, res) => {
+  const { id } = req.params
+
+  try {
+    const updatedThought = await Thought.findByIdAndUpdate(id, { $inc: { hearts: 1 } })
```
Comment (KarolCislowski) [inline server.js:61]: 💯

**Review by KarolCislowski — APPROVED** [review]
Nice API but remember to always provide documentation on how to use your apps

**Pull request closed without merging**